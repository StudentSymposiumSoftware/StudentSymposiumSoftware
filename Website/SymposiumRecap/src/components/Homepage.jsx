import "./Homepage.css"

function HomepageComponent() {
    return (
        <div id="homepage-container">
            <h2>Welcome to the Symposium Recap site !!</h2>
            <h3>Please select the needed link above to route to your component</h3>
        </div>
    )
}

export default HomepageComponent
  