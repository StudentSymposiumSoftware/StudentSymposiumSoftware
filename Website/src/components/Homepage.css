body {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    margin: 0;
    font-family: Arial, sans-serif;
    background-color: #f5f5f5;
}

#homepage-container {
    text-align: center;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.abstract-display {
    background-color: white;
    padding: 20px;
    border-radius: 15px;
    width: 50%;
    min-width: 300px;
    max-width: 600px;
    margin: 20px auto;
    text-align: left;
    color: black;
}