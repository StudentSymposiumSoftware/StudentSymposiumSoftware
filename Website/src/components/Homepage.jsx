import "./Homepage.css";
import { useEffect, useState } from "react";

function HomepageComponent(props) {
    const [randomAbstract, setRandomAbstract] = useState(null);

    useEffect(() => {
        const updateAbstract = () => {
            if (props.data && props.data.length > 0) {
                const randomIndex = Math.floor(Math.random() * props.data.length);
                setRandomAbstract(props.data[randomIndex]);
            }
        };

        updateAbstract();
        const interval = setInterval(updateAbstract, 5000);

        return () => clearInterval(interval);
    }, [props.data]);

    return (
        <div id="homepage-container">
            <h2>Welcome to the Symposium Recap site !!</h2>
            <h3>Please select the needed link above to route to your component</h3>
            {randomAbstract && (
                <div className="abstract-display">
                    <h4>{randomAbstract.title}</h4>
                    <p><strong>Author:</strong> {randomAbstract.author}</p>
                    <p><strong>Professor:</strong> {randomAbstract.professor}</p>
                    <p><strong>School:</strong> {randomAbstract.school}</p>
                    <p><strong>Major:</strong> {randomAbstract.Major_1}</p>
                    <p><strong>Category:</strong> {randomAbstract.category}</p>
                    <p><strong>Abstract Number:</strong> {randomAbstract.abstractNumber}</p>
                </div>
            )}
        </div>
    );
}

export default HomepageComponent;
