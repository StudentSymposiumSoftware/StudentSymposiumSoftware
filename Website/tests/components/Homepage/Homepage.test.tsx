import React from 'react';
import { render } from "@testing-library/react";
import { expect, it } from "vitest";

import Homepage from '../../../src/components/Homepage';

const mockData = [
    { title: "Abstract 1", author: "Author A", professor: "Prof. X", school: "School Y", Major_1: "Major Z", category: "Category 1", abstractNumber: "001" },
    { title: "Abstract 2", author: "Author B", professor: "Prof. Y", school: "School X", Major_1: "Major A", category: "Category 2", abstractNumber: "002" }
];

it('Homepage renders correctly and updates abstract', () => {
    const tree = render(<Homepage data={mockData} />);
    expect(tree).toMatchSnapshot();
});
