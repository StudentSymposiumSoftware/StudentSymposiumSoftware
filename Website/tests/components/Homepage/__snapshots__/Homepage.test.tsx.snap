// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage renders correctly and updates abstract 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="homepage-container"
      >
        <h2>
          Welcome to the Symposium Recap site !!
        </h2>
        <h3>
          Please select the needed link above to route to your component
        </h3>
        <div
          class="abstract-display"
        >
          <h4>
            Abstract 2
          </h4>
          <p>
            <strong>
              Author:
            </strong>
             
            Author B
          </p>
          <p>
            <strong>
              Professor:
            </strong>
             
            Prof. Y
          </p>
          <p>
            <strong>
              School:
            </strong>
             
            School X
          </p>
          <p>
            <strong>
              Major:
            </strong>
             
            Major A
          </p>
          <p>
            <strong>
              Category:
            </strong>
             
            Category 2
          </p>
          <p>
            <strong>
              Abstract Number:
            </strong>
             
            002
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="homepage-container"
    >
      <h2>
        Welcome to the Symposium Recap site !!
      </h2>
      <h3>
        Please select the needed link above to route to your component
      </h3>
      <div
        class="abstract-display"
      >
        <h4>
          Abstract 2
        </h4>
        <p>
          <strong>
            Author:
          </strong>
           
          Author B
        </p>
        <p>
          <strong>
            Professor:
          </strong>
           
          Prof. Y
        </p>
        <p>
          <strong>
            School:
          </strong>
           
          School X
        </p>
        <p>
          <strong>
            Major:
          </strong>
           
          Major A
        </p>
        <p>
          <strong>
            Category:
          </strong>
           
          Category 2
        </p>
        <p>
          <strong>
            Abstract Number:
          </strong>
           
          002
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
